Runtime error with bitcoind backend

I've been trying out the new `bitcoind` backend mode, and I've been having some issues. 

After starting `bitcoind` (and ensuring it syncs completely before doing anything else), `lnd` gives me this errror:

```
2018-01-18 11:02:01.643 [INF] LTND: Version 0.3.0-alpha
2018-01-18 11:02:01.643 [INF] CHDB: Checking for schema update: latest_version=0, db_version=0
2018-01-18 11:02:01.643 [INF] LTND: Primary chain is set to: bitcoin
2018-01-18 11:02:01.643 [INF] LTND: Initializing bitcoind backed fee estimator
2018-01-18 11:02:02.337 [INF] LNWL: Opened wallet
panic: runtime error: index out of range

goroutine 10 [running]:
github.com/lightningnetwork/lnd/vendor/github.com/lightninglabs/gozmq.connFromAddr(0x0, 0x0, 0x0, 0x0, 0x3, 0xc4201fc780)
        /home/torkel/goprojects/src/github.com/lightningnetwork/lnd/vendor/github.com/lightninglabs/gozmq/zmq.go:286 +0x467
github.com/lightningnetwork/lnd/vendor/github.com/lightninglabs/gozmq.Subscribe(0x0, 0x0, 0xc4201bbeb0, 0x2, 0x2, 0x5f5e100, 0x0, 0x0, 0xc420066700)
        /home/torkel/goprojects/src/github.com/lightningnetwork/lnd/vendor/github.com/lightninglabs/gozmq/zmq.go:312 +0x4d
github.com/lightningnetwork/lnd/vendor/github.com/roasbeef/btcwallet/chain.(*BitcoindClient).socketHandler(0xc4201da540)
        /home/torkel/goprojects/src/github.com/lightningnetwork/lnd/vendor/github.com/roasbeef/btcwallet/chain/bitcoind.go:515 +0xf3
created by github.com/lightningnetwork/lnd/vendor/github.com/roasbeef/btcwallet/chain.(*BitcoindClient).Start
        /home/torkel/goprojects/src/github.com/lightningnetwork/lnd/vendor/github.com/roasbeef/btcwallet/chain/bitcoind.go:368 +0x160
```

My `lnd.conf` looks like this:
```
[Application Options]
no-macaroons=true
noencryptwallet=true

[Bitcoin]
bitcoin.active=true
bitcoin.testnet=true
bitcoin.node=bitcoind

[bitcoind]
bitcoind.rpcuser=xxx
bitcoind.rpcpass=xxx
```

My `bitcoin.conf` looks like this:
```
testnet=1
rpcuser=xxx
rpcpassword=xxx
server=1
txindex=1
zmqpubrawblock=tcp://127.0.0.1:29000
zmqpubrawtx=tcp://127.0.0.1:29000
````

I'm running Ubuntu 17.10, with Go version 1.9.2.

--- a/config.go
+++ b/config.go
@@ -621,12 +621,29 @@ func parseRPCParams(cConfig *chainConfig, nodeConfig interface{}, net chainCode,
 	// specified, we can return.
 	switch conf := nodeConfig.(type) {
 	case *btcdConfig:
-		if conf.RPCUser != "" || conf.RPCPass != "" {
+		// If both RPCUser and RPCPass are set, we assume those
+		// credentials are good to use.
+		if conf.RPCUser != "" && conf.RPCPass != "" {
 			return nil
 		}
+		// If only ONE of RPCUser or RPCPass is set, we assume the
+		// user did that unintentionally.
+		if conf.RPCUser != "" || conf.RPCPass != "" {
+			return fmt.Errorf("please set both or neither of " +
+				"btcd.rpcuser and btcd.rpcpass")
+		}
 	case *bitcoindConfig:
-		if conf.RPCUser != "" || conf.RPCPass != "" || conf.ZMQPath != "" {
+		// If all of RPCUser, RPCPass, and ZMQPath are set, we assume
+		// those parameters are good to use.
+		if conf.RPCUser != "" && conf.RPCPass != "" && conf.ZMQPath != "" {
 			return nil
+		}
+		// If only one or two of the parameters are set, we assume the
+		// user did that unintentionally.
+		if conf.RPCUser != "" || conf.RPCPass != "" || conf.ZMQPath != "" {
+			return fmt.Errorf("please set all or none of " +
+				"bitcoind.rpcuser, bitcoind.rpcpass, and " +
+				"bitcoind.zmqpath")
 		}
 	}
 

